prow/spyglass/podlogartifact_fetcher.go

```diff
@@ -34,11 +36,23 @@ func NewPodLogArtifactFetcher(ja jobAgent) *PodLogArtifactFetcher {
 }
 
 // artifact constructs an artifact handle for the given job build
-func (af *PodLogArtifactFetcher) Artifact(_ context.Context, jobName, buildID string, sizeLimit int64) (api.Artifact, error) {
-
-	podLog, err := NewPodLogArtifact(jobName, buildID, sizeLimit, af.jobAgent)
+func (af *PodLogArtifactFetcher) Artifact(_ context.Context, key string, artifactName string, sizeLimit int64) (api.Artifact, error) {
+	common.KeyToJob(key)
+	jobName, buildID, err := common.KeyToJob(key)
+	if err != nil {
+		return nil, fmt.Errorf("could not derive job: %v", err)
+	}
+	containerName := containerName(artifactName)
+	podLog, err := NewPodLogArtifact(jobName, buildID, artifactName, containerName, sizeLimit, af.jobAgent)
 	if err != nil {
 		return nil, fmt.Errorf("error accessing pod log from given source: %w", err)
 	}
 	return podLog, nil
 }
+
+func containerName(artifactName string) string {
+	if artifactName == "build-log.txt" {
+		return kube.TestContainerName
+	}
+	return artifactName[0 : len(artifactName)-len("-build-log.txt")]
```

Comment:
Is there something simpler (and safer) that can do the same? (Maybe a "replace trailing `"-build-log.txt"` with `""`"?)

[Hint, hint](https://golang.org/pkg/strings/#TrimSuffix)